### DISCUSSION TITLE
Failed to compile file

### DISCUSSION THREAD

**Discussion by 4867896** [IchthysMaranatha/asterisk-chan-quectel#40] [Q&A]
I tried to edit it in the debian chroot environment on the Raspberry Pi, but I couldn't compile it on the ARM. Can you help me compile the library files for the ARM and ARM64 platforms. thanks

**Comment by 4867896** [comment]
./configure --with-astversion=18.11 --with-asterisk=/tmp/asterisk-master/include/  DESTDIR=/tmp/asterisk-master/
make 

some err
.......
/tmp/asterisk-master/include/asterisk/select.h:43:2: error: unknown type name ‘TYPEOF_FD_SET_FDS_BITS’
  TYPEOF_FD_SET_FDS_BITS fds_bits[ast_FDMAX / 8 / SIZEOF_FD_SET_FDS_BITS]; /* 32768 bits */
  ^~~~~~~~~~~~~~~~~~~~~~
/tmp/asterisk-master/include/asterisk/select.h:43:50: error: ‘SIZEOF_FD_SET_FDS_BITS’ undeclared here (not in a function)
  TYPEOF_FD_SET_FDS_BITS fds_bits[ast_FDMAX / 8 / SIZEOF_FD_SET_FDS_BITS]; /* 32768 bits */
                                                  ^~~~~~~~~~~~~~~~~~~~~~
In file included from app.c:23:0:
chan_quectel.h:29:28: fatal error: alsa/asoundlib.h: No such file or directory

                            ^
compilation terminated.
Makefile:76: recipe for target 'app.o' failed
make: *** [app.o] Error 1

**Reply by IchthysMaranatha** [reply]
Try again after installing alsa first

**Reply by 4867896** [reply]
gcc -g -O2 -O6 -Wno-old-style-declaration -I. -std=gnu99 -DAST_MODULE_SELF_SYM=__internal_chan_quectel_self -D_GNU_SOURCE -I/tmp/asterisk-master/include/ -I/usr/include -DHAVE_CONFIG_H  -fvisibility=hidden -fPIC -Wall -Wextra -MD -MT app.o -MF .app.o.d -MP  -o app.o -c app.c
In file included from ast_config.h:10:0,
                 from app.c:2:
/tmp/asterisk-master/include/asterisk.h:21:33: fatal error: asterisk/autoconfig.h: No such file or directory
 #include "asterisk/autoconfig.h"
                                 ^
compilation terminated.
Makefile:76: recipe for target 'app.o' failed
make: *** [app.o] Error 1

i have installed   linux-headers-4.9.0-13-all  
that is new error

**Comment by IchthysMaranatha** [comment]
You have to build asterisk first, have you done that? 
Have compiled on Pi without issues. Adapt for pi from this https://github.com/IchthysMaranatha/asterisk-chan-quectel/discussions/13

**Reply by 4867896** [reply]
OK I know thankyou very much
